### PR TITLE
Add additional startup logging

### PR DESCRIPTION
**Pull request by sgraband**
#### What it does

- Add log message with timestamp of finished applicaiton loading
- Add timestamp from frontend start to sync, load and start plugin durations
- Split sync duration in waiting for backend deployment and sync
- This prevents the sync time to be longer because the backend is not yet ready

Contributed on behalf of STMicroelectronics
Signed-off-by: Simon Graband <sgraband@eclipsesource.com>
#### Review checklist

- [ ] As an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

File: packages/plugin-ext/src/hosted/browser/hosted-plugin.ts [hunk 1]
```diff
@@ -712,6 +715,17 @@ export class HostedPluginSupport {
         };
     }
 
+    protected logDeploymentMeasurement(measurement: () => number): void {
```
Comment (sdirix) [inline packages/plugin-ext/src/hosted/browser/hosted-plugin.ts:718]: This is exactly the same code as the other log message just with a different text. Shouldn't we just make the text configurable in a better way?

File: packages/core/src/browser/frontend-application.ts [hunk 2]
```diff
@@ -229,8 +229,9 @@ export class FrontendApplication {
         const startupElem = this.getStartupIndicator(host);
         if (startupElem) {
             return new Promise(resolve => {
-                window.requestAnimationFrame(() => {
+                window.requestAnimationFrame(async () => {
                     startupElem.classList.add('theia-hidden');
+                    console.log('Finished loading application after: ' + (performance.now() / 1000).toFixed(3) + ' seconds');
```
Comment (sdirix) [inline packages/core/src/browser/frontend-application.ts:234]:
```suggestion
                    console.log(`Finished loading frontend application after ${(performance.now() / 1000).toFixed(3)} seconds`);
```

File: packages/plugin-ext/src/hosted/browser/hosted-plugin.ts [hunk 3]
```diff
@@ -712,6 +715,17 @@ export class HostedPluginSupport {
...
+        const duration = measurement();
+        if (duration === Number.NaN) {
+            // Measurement was prevented by native API, do not log NaN duration
+            return;
+        }
+
+        const timeStamp = (performance.now() / 1000).toFixed(3) + ' s from frontend start';
```
Comment (sdirix) [inline packages/plugin-ext/src/hosted/browser/hosted-plugin.ts:725]:
```suggestion
        const timeFromFrontendStart = `Finished ${(performance.now() / 1000).toFixed(3)} after frontend start`;
```

**Review by sdirix — COMMENTED** [review]
I tried to reformulate the texts to make them perfectly clear even on first-time reading. What do you think?

File: packages/plugin-ext/src/hosted/browser/hosted-plugin.ts (hunk 1 above)
Comment (sgraband) [reply packages/plugin-ext/src/hosted/browser/hosted-plugin.ts]: @sdirix What do you think about the current solution? I moved the text creation completely to the call and made a new method that calculates the pluginCount string.

**Review by sdirix — APPROVED** [review]

**Comment by sdirix** [comment]
Merged upstream with https://github.com/eclipse-theia/theia/pull/10116

**Pull request closed without merging**